# Description: Run /etc/X11/Xsession.d/.
# Ubuntu-Specific: /etc/X11/Xsession.d is a Debianism/Ubuntuism
--- gdm-2.26.1git20090717/data/Xsession.in	2009-07-20 09:08:56.824501675 +0200
+++ gdm-2.26.1git20090717.new/data/Xsession.in	2009-07-20 09:09:04.884502723 +0200
@@ -186,6 +186,15 @@
     done
 fi
 
+# run /etc/X11/Xsession.d/
+if [ -d /etc/X11/Xsession.d ]; then
+    for i in `ls /etc/X11/Xsession.d/ ; do
+        if [ -r "$i" ] && expr "$i" : '^[[:alnum:]_-]\+$' > /dev/null; then
+	    . "$i"
+        fi
+    done
+fi
+
 if [ "x$command" = "xcustom" ] ; then
   if [ -x "$HOME/.xsession" ]; then
     command="$HOME/.xsession"
